EntityLabelLib-Bukkit-NmsImpl/src/main/java/io/github/arcaneplugins/entitylabellib/bukkit/nms/NmsLabelHandler.java

```diff
@@ -35,6 +36,24 @@
 
 public class NmsLabelHandler extends LabelHandler implements Listener {
 
+    private static final EntityDataAccessor<Boolean> LABEL_VISIBLE;
+    private static final EntityDataAccessor<Optional<net.minecraft.network.chat.Component>> LABEL;
+
+    static {
+        Class<?> entityClass = net.minecraft.world.entity.Entity.class;
+        try {
+            Field labelVisible = entityClass.getDeclaredField("DATA_CUSTOM_NAME_VISIBLE");
+            labelVisible.setAccessible(true);
+            LABEL_VISIBLE = (EntityDataAccessor<Boolean>) labelVisible.get(null);
+
+            Field label = entityClass.getDeclaredField("DATA_CUSTOM_NAME");
+            label.setAccessible(true);
+            LABEL = (EntityDataAccessor<Optional<net.minecraft.network.chat.Component>>) label.get(null);
```

Comment:
This would modify the values server-side instead of client-side, no?

Reply:
No because the only one receiving the packet is the target player. The client doesn't set the entity data, the server does.